packages/itwin/tree-widget/src/tree-widget-react/components/trees/common/components/TreeRenderer.tsx

```diff
@@ -25,7 +25,10 @@ export function TreeRenderer({ rootNodes, onNodeClick, expandNode, visibilityBut
         onNodeClick={onNodeClick}
         expandNode={expandNode}
         rootNodes={rootNodes}
-        actionsRenderer={visibilityButtonProps ? (node) => <TreeItemVisibilityButton {...visibilityButtonProps} node={node} /> : undefined}
+        actions={[
+          ...(visibilityButtonProps ? [createVisibilityAction(visibilityButtonProps)] : []),
```

Comment:
Can we move code related to visibility buttons into `VisibilityTreeRenderer`?

Reply:
The only issue would be that filter button would go first then visibility. If this order is fine, have the changes ready.

Reply:
I think we could duplicate filter creation in `VisibilityTreeRenderer`. This might result in use having 3 components: `TreeRenderer`, `VisibilityTreeRenderer` and `BaseTreeRenderer` (internal) but would allow to keep separate API for visibility and non-visibility trees. It could look like:

```tsx
function BaseTreeRenderer(props) {
  // setup and render tree
  return <TreeComp
    actions={props.actions}
  />
}

function TreeRenderer(props) {
  return <BaseTreeRenderer 
    actions={]createFillterAction(), ...props.consumerActions]}
  />;
}

function VisibilityTreeRenderer(props) {
  return <BaseTreeRenderer 
    actions={[createVisibilityAction(), createFilterAction(), ...props.consumerActions]}
  />
}
```

Reply:
Changed and added a way to provide actions to all base trees, which was missing.